datalad/support/gitrepo.py

```diff
@@ -671,21 +674,174 @@ def git_remote_update(self, name='', verbose=False):
         self._git_custom_command('', 'git remote %s update %s' % (name, v),
                                  expect_stderr=True)
 
-    def git_fetch(self, name, options=''):
+    # TODO: centralize all the c&p code in fetch, pull, push
+    # TODO: document **kwargs passed to gitpython
+    def fetch(self, remote=None, refspec=None, progress=None, all=False, **kwargs):
+        # TODO: options=> **kwargs):
+        """
+        """
+        # Note: Apparently there is no explicit (fetch --all) in gitpython,
+        #       but fetch is always bound to a certain remote instead.
+        #       Therefore implement it on our own:
+        if remote is None:
+            if refspec is not None:
+                # conflicts with using tracking branch or fetch all remotes
+                # For now: Just fail.
+                # TODO: May be check whether it fits to tracking branch
+                raise ValueError("refspec specified without a remote. (%s)" %
+                                  refspec)
+            if all:
+                remotes_to_fetch = self.repo.remotes
+            else:
+                # No explicit remote to fetch.
+                # => get tracking branch:
+                tb = self.repo.active_branch.tracking_branch().name
+                if tb:
+                    parts = tb.split('/')
+                    assert len(parts) == 2
```

Comment:
nope -- branch names could have '/' in them.  We do e.g. in pymvpa.   just `split('/', 1)` and since code below already hard codes assumption of having two pieces, you can `tb_remote, tb_branch = tb.split('/', 1)` I guess, may be with `assert '/' in tb, "tracking branch %s must have had a /" % tb"` before